services/construction/contract_call_data.go

```diff
@@ -80,6 +89,33 @@ func ConstructContractCallDataGeneric(methodSig string, methodArgs interface{})
 	}
 }
 
+// preprocessArgs converts methodArgs to a string value if methodSig is an empty string.
+// We are calling a contract written with fallback pattern, which has no method signature.
+func preprocessArgs(methodSig string, methodArgs interface{}) (interface{}, error) {
+	if methodSig != "" && methodSig != NoMethodSig {
+		return methodArgs, nil
+	}
+
+	switch args := methodArgs.(type) {
+	case []interface{}:
+		if len(args) == 1 {
+			argStr, isStrVal := args[0].(string)
+			if !isStrVal {
+				return nil, fmt.Errorf("failed to convert method arg \"%T\" to string", args[0])
+			}
+			return argStr, nil
+		}
+		return methodArgs, nil
+	case []string:
```

Comment:
do we need to handle this case?

Reply:
For this specific project, no. Doing it for generalization.